### PR TITLE
add keras2 Conv1D layer and test

### PR DESCRIPTION
**Pull request by simanw304**

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                         val strides: Int = 1,
+                         val padding: String = "valid",
+                         val data_format: String = "channels_last",
+                         override val activation: KerasLayer[Tensor[T], Tensor[T], T] = null,
+                         val use_bias: Boolean = true,
+                         val kernel_initializer: InitializationMethod = Xavier,
+                         val bias_initializer: InitializationMethod = Zeros,
+                         val kernel_Regularizer: Regularizer[T] = null,
+                         val bias_Regularizer: Regularizer[T] = null,
+                         val activity_Regularizer: Regularizer[T] = null,
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala]: remove activity_Regularizer

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  *                             the output of the layer (its "activation").
+  * @param inputShape A Single Shape, does not include the batch dimension.
+  * @tparam T Numeric type of parameter(e.g. weight, bias). Only support float/double now.
+  */
+class Conv1D[T: ClassTag](
+                         val filters: Int,
+                         val kernel_size: Int,
+                         val strides: Int = 1,
+                         val padding: String = "valid",
+                         val data_format: String = "channels_last",
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala]: remove data_format

**Comment by simanw304** [comment]
jenkins passed http://10.239.47.211:18888/job/ZOO-PR-Validation/602/

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import com.intel.analytics.zoo.pipeline.api.Net
+import com.intel.analytics.zoo.pipeline.api.keras.layers.utils.KerasUtils
+import com.intel.analytics.zoo.pipeline.api.keras.{layers => klayers1}
+
+import scala.reflect.ClassTag
+
+/**
+ * Applies convolution operator for filtering neighborhoods of 1-D inputs.
+ * You can also use Conv1D as an alias of this layer.
+ * The input of this layer should be 3D.
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala]: Add more description:
```
Input shape

3D tensor with shape: (samples, steps, input_dim) or (samples, steps, channels).

Output shape

3D tensor with shape: (samples, new_steps, filters). steps value might have changed due to padding.
```

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ * @param bias_initializer Initializer for the bias vector.
+ * @param kernel_Regularizer Regularizer function applied to
+ *                           the `kernel` weights matrix Default is null.
+ * @param bias_Regularizer Regularizer function applied to the bias vector. Default is null.
+ * @param inputShape A Single Shape, does not include the batch dimension.
+ * @tparam T Numeric type of parameter(e.g. weight, bias). Only support float/double now.
+ */
+class Conv1D[T: ClassTag](
+   val filters: Int,
+   val kernel_size: Int,
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala]: change to `kernelSize`

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+   val strides: Int = 1,
+   val padding: String = "valid",
+   override val activation: KerasLayer[Tensor[T], Tensor[T], T] = null,
+   val use_bias: Boolean = true,
+   val kernel_initializer: InitializationMethod = Xavier,
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1D.scala]: kernelInitializer

File: zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1DSpec.scala [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    checkOutputAndGrad(seq.asInstanceOf[AbstractModule[Tensor[Float], Tensor[Float], Float]],
+      kerasCode, weightConverter)
+  }
+
+  "Conv1D without bias" should "be the same as Keras" taggedAs(Keras2Test) in {
+    val kerasCode =
+      """
+        |input_tensor = Input(shape=[20, 32])
+        |input = np.random.random([2, 20, 32])
+        |output_tensor = Convolution1D(32, 4, activation="relu", bias=False,
```
Comment (zhichao-li) [inline zoo/src/test/scala/com/intel/analytics/zoo/pipeline/api/keras2/layers/Conv1DSpec.scala]: change to `Conv1D` ?

**Comment by simanw304** [comment]
jenkins passed http://10.239.47.211:18888/job/ZOO-PR-Validation/612/

**Pull request merged**